pkg/rancher-desktop/config/settings.ts

```diff
@@ -474,114 +473,142 @@ function updateSettings(settings: Settings) {
 // Imported from dashboard/config/settings.js
 // Setting IDs
 export const SETTING = { PL_RANCHER_VALUE: 'rancher' };
+const REGISTRY_PATH_PROFILE = ['SOFTWARE', 'Rancher Desktop', 'Profile'];
 
 /**
- * Read and validate the json formatted deployment profile.
- * Returns the system profile if it exists, otherwise the user profile if it exists.
- * @returns A settings profile stripped of all fields not found in the schema.
+ * Read and validate deployment profiles, giving system level profiles
+ * priority over user level profiles.  If the system directory contains a
+ * defaults or locked profile, the user directory will not be read.
+ * @returns undefined, or type validated defaults and locked deployment profiles.
  */
-export function readDeploymentProfile() {
-  let profileData;
-  let profile;
+export function readDeploymentProfiles() {
+  let defaults;
+  let locked;
 
   switch (os.platform()) {
   case 'win32':
-    try {
-      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
-    } catch {
-      try {
-        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
-      } catch {}
+    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+
+      if (registryKey !== null) {
+        defaults = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Defaults']);
+        locked = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Locked']);
+        nativeReg.closeKey(registryKey);
+        if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+          break;
+        }
+      }
     }
     break;
   case 'linux':
-    try {
-      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_JSON), 'utf8');
-    } catch {
+    for (const rootPath of [paths.deploymentProfileSystem, paths.deploymentProfileUser]) {
       try {
-        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_JSON), 'utf8');
+        const defaultsData = fs.readFileSync(join(rootPath, 'defaults.json'), 'utf8');
+
+        defaults = JSON.parse(defaultsData);
       } catch {}
-    }
-    if (typeof profileData !== 'undefined') {
-      profile = JSON.parse(profileData);
+      try {
+        const lockedData = fs.readFileSync(join(rootPath, 'locked.json'), 'utf8');
+
+        locked = JSON.parse(lockedData);
```

Comment:
If there's no `lockedData`, `locked` should be set to `{}`

So you need something like
```
} catch(ex: any) {
  if (ex.code === 'ENOENT') {
    locked = {};
  }
}
```

Reply:
Currently, if the defaults or locked file don't exist, the value will be undefined.  If the file exists, but is an empty struct, or has no valid fields, it will be {}.  Are you saying you want the defaults and locked values to be {} if the files do not exist?

Reply:
Definitely, if the locked file doesn't exist, I want `locked = {}` -- this means no fields are locked, and is easier to process than if it's set to `null` or `undefined`

If the `defaults.X` file doesn't exist, there's no point looking for a `locked` file -- either move from `system` to `user`, or if processing `user`, return `undefined`

Reply:
> If the `defaults.X` file doesn't exist, there's no point looking for a `locked` file -- either move from `system` to `user`, or if processing `user`, return `undefined`<

If I want to lock down the `imageAllowList` I thought I would put the list in the locked file, and have no need for the defaults file.  Is that not how this is supposed to work?

Reply:
@dpmerrill is correct, you can have a "locked" profile without a "defaults" profile. If either exists in the "system" context, then the user context is ignored completely.

We will never mix profiles from different contexts; if any setting for "system" exists, **all** the "user" settings will be ignored.

Reply:
done.  `defaults` and `locked` will be empty objects (`{}`) if there is no data.

Reply:
I thought we were going with what I proposed at https://github.com/rancher-sandbox/rancher-desktop/issues/3862#issuecomment-1414522128 -- the `lockedSettings` is an object of paths pointing to booleans that indicate whether a setting is locked or not, but the actual values live only in the current settings object.